src/main/java/com/rsr5/gocd/riemann_notifier/PipelineDetailsPopulator.java

```diff
@@ -26,11 +41,42 @@ private JsonObject mergeInPipelineInstanceDetails(String attributeName,
         return json;
     }
 
-    protected JsonElement downloadPipelineDetails(String pipelineName) throws IOException {
-        String path = "/go/api/pipelines/" + pipelineName + "/history";
-        JsonObject json = accessor.get(path).getAsJsonObject();
-        JsonArray pipelines = json.get("pipelines").getAsJsonArray();
-        return pipelines.get(0);
+    /**
+     * Performs a depth-first search of the artifacts, returning a JsonObject
+     * translation of the first artifact named "results.xml".
+     */
+    protected JsonElement findJunitResults(JsonArray artifacts) throws IOException {
+        for (int i = 0; i < artifacts.size(); i++) {
+            JsonObject artifact = artifacts.get(i).getAsJsonObject();
```

Comment:
Looks like `JsonArray` is iterable, so this could use a foreach loop instead to free you from the index tracking:
```suggestion
        for (JsonElement element : artifacts) {
            JsonObject artifact = element.getAsJsonObject();
```